compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -13605,7 +13605,7 @@ private STNode parseErrorArgListMatchPattern(ParserRuleContext context) {
                 return parseMatchPattern();
             case VAR_KEYWORD:
                 STNode varKeyword = consume();
-                STNode variableName = parseVariableName();
+                STNode variableName = parseBindingPattern();
```

Comment:
Should only allow the capture binding pattern here. Can use `createCaptureOrWildcardBP` method to create capture binding pattern from var name
```suggestion
                STNode variableName = createCaptureOrWildcardBP(parseVariableName());
```

Reply:
Fixed